client/history/client.go

```diff
@@ -821,16 +822,24 @@ func (c *clientImpl) GetReplicationMessages(
 		req.Tokens = append(req.Tokens, token)
 	}
 
+	// preserve some time to return partial of the result if context is timing out
+	now := time.Now()
+	deadline, ok := ctx.Deadline()
+	if !ok {
+		deadline = now.Add(c.timeout)
+	}
+	requestTimeout := time.Duration(math.Ceil(float64(deadline.Sub(now)) * 0.95))
+	requestContext, cancel := context.WithTimeout(context.Background(), requestTimeout)
```

Comment:
nit: not sure if using `ctx` instead of `context.Backgroud()` can preserve more information (like for tracing) here?

Reply:
Good point. Will change to the original context